### PR TITLE
Fix seed get error

### PR DESCRIPTION
**Pull request by DrownFish19**
### PR types

Bug fixes

### PR changes

APIs

### Description

* fix seed get error, add if before get

### CODE REVIEW

File: llm/gpt-3/modeling.py [hunk 1]
```diff
@@ -285,7 +285,7 @@ def core_attn(self, q, k, v, attn_mask=None, output_attentions=False):
             weights = incubate.softmax_mask_fuse_upper_triangle(product)
 
         if self.config.hidden_dropout_prob:
-            if self.training:
+            if self.training and "local_seed" in get_rng_state_tracker().states_:
```
Comment (ZHUI) [inline llm/gpt-3/modeling.py]: https://github.com/PaddlePaddle/PaddleNLP/blob/48db6d42e2c44c53207ca655ee69de37014a6eeb/paddlenlp/trainer/trainer_utils.py#L78-L80

Comment (ZHUI) [inline llm/gpt-3/modeling.py]:
```python
        if self.training:
            with get_rng_state_tracker().rng_state(current_seed):
                if not self.config.use_fused_dropout_add:
                    tgt = residual + self.dropout1(tgt)
                else:
                    tgt = self.fused_dropout_add1(tgt, residual)
        else:
            if not self.config.use_fused_dropout_add:
                tgt = residual + self.dropout1(tgt)
            else:
                tgt = self.fused_dropout_add1(tgt, residual)
```

```python
def seed_guard_context(name=None):
      ctx_manager = contextlib.nullcontext()
        pass

          with seed_guard_context("current_seed"):
              if not self.config.use_fused_dropout_add:
                  tgt = residual + self.dropout1(tgt)
              else:
                  tgt = self.fused_dropout_add1(tgt, residual)
```

Comment (ZHUI) [reply llm/gpt-3/modeling.py]: https://github.com/PaddlePaddle/PaddleNLP/blob/1b3973ca493e3c816db31902ca27d42accfeaddc/paddlenlp/trainer/trainer.py#L1503C13-L1503C51

File: llm/gpt-3/modeling.py [hunk 2]
```diff
@@ -897,9 +888,7 @@ def prepare_inputs_for_generation(self, input_ids, use_cache=False, cache=None,
 
     @staticmethod
     def prepare_attention_mask_for_generation(input_ids, pad_token_id, eos_token_id):
-        is_pad_token_in_inputs_ids = (pad_token_id is not None) and paddle.any(
-            input_ids == pad_token_id
-        ).numpy().item()
+        is_pad_token_in_inputs_ids = (pad_token_id is not None) and float(paddle.any(input_ids == pad_token_id))
```
Comment (DrownFish19) [inline llm/gpt-3/modeling.py:891]: 针对paddle2.5.1的0-D warning进行修改

File: llm/gpt-3/utils.py [hunk 3]
```diff
@@ -328,7 +328,7 @@ def prediction_step(
 
         preds = model.generate(
             input_ids=inputs["input_ids"],
-            attention_mask=inputs["attention_mask"],
+            attention_mask=inputs["attention_mask"] if "attention_mask" in inputs else None,
```
Comment (DrownFish19) [inline llm/gpt-3/utils.py:331]: inputs中不存在"attention_mask"，此处增加判断

**Review by ZHUI — APPROVED** [review]
LGTM

**Review by ZHUI — APPROVED** [review]
LGTM
